cmd/webhook/main.go

```diff
@@ -31,78 +31,104 @@ import (
 	}
-	flag.StringVar(&admissionListenAddress, "listen-address", ":9876", "The address on which the MutatingWebhook will listen on")
-	flag.StringVar(&admissionTLSCertPath, "tls-cert-path", "/tmp/cert/cert.pem", "The path of the TLS cert for the MutatingWebhook")
-	flag.StringVar(&admissionTLSKeyPath, "tls-key-path", "/tmp/cert/key.pem", "The path of the TLS key for the MutatingWebhook")
-	flag.StringVar(&caBundleFile, "ca-bundle", "", "path to a file containing all PEM-encoded CA certificates (will be used instead of the host's certificates if set)")
-	flag.StringVar(&namespace, "namespace", "kubermatic", "The namespace where the webhooks will run")
+	flag.StringVar(&opt.admissionListenAddress, "listen-address", ":9876", "The address on which the MutatingWebhook will listen on")
+	flag.StringVar(&opt.admissionTLSCertPath, "tls-cert-path", "/tmp/cert/cert.pem", "The path of the TLS cert for the MutatingWebhook")
+	flag.StringVar(&opt.admissionTLSKeyPath, "tls-key-path", "/tmp/cert/key.pem", "The path of the TLS key for the MutatingWebhook")
+	flag.StringVar(&opt.caBundleFile, "ca-bundle", "", "path to a file containing all PEM-encoded CA certificates (will be used instead of the host's certificates if set)")
+	flag.StringVar(&opt.namespace, "namespace", "kubermatic", "The namespace where the webhooks will run")
+	flag.StringVar(&opt.workerClusterKubeconfig, "worker-cluster-kubeconfig", "", "Path to kubeconfig of worker/user cluster where machines and machinedeployments exist")
 
 	// OSM specific flags
-	flag.BoolVar(&useOSM, "use-osm", false, "osm controller is enabled for node bootstrap")
+	flag.BoolVar(&opt.useOSM, "use-osm", false, "osm controller is enabled for node bootstrap")
 
 	flag.Parse()
-	kubeconfig = flag.Lookup("kubeconfig").Value.(flag.Getter).Get().(string)
-	masterURL = flag.Lookup("master").Value.(flag.Getter).Get().(string)
+	opt.kubeconfig = flag.Lookup("kubeconfig").Value.(flag.Getter).Get().(string)
+	opt.masterURL = flag.Lookup("master").Value.(flag.Getter).Get().(string)
 
-	if caBundleFile != "" {
-		if err := util.SetCABundleFile(caBundleFile); err != nil {
+	if opt.caBundleFile != "" {
+		if err := util.SetCABundleFile(opt.caBundleFile); err != nil {
 			klog.Fatalf("-ca-bundle is invalid: %v", err)
 		}
 	}
 
-	cfg, err := clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
-	if err != nil {
-		klog.Fatalf("error building kubeconfig: %v", err)
-	}
-
+	// Add osmv1alpha1 to scheme
 	if err := osmv1alpha1.AddToScheme(scheme.Scheme); err != nil {
 		klog.Fatalf("failed to add osmv1alpha1 api to scheme: %v", err)
 	}
 
-	client, err := ctrlruntimeclient.New(cfg, ctrlruntimeclient.Options{})
+	cfg, err := clientcmd.BuildConfigFromFlags(opt.masterURL, opt.kubeconfig)
+	if err != nil {
+		klog.Fatalf("error building kubeconfig: %v", err)
+	}
+
+	client, err := ctrlruntimeclient.New(cfg, ctrlruntimeclient.Options{
+		Scheme: scheme.Scheme,
```

Comment:
This isn't needed, just leave `ctrlruntimeclient.Options{}` empty as it was before.